doc/develop/api/design_guidelines.rst

```diff
@@ -120,3 +127,31 @@ not implemented or optional:
   requested in the call cannot be satisfied by the implementation the call shall
   return ``-ENOTSUP``. (For example, a request for a level-triggered GPIO interrupt on
   hardware that supports only edge-triggered interrupts)
+
+The table below shows the error code conventions based on device driver use cases, but they can also be applied to other kernel components.
```

Comment:
What's this information is addressing specifically? Limiting the error codes used by some APIs? Unclear to me sorry.

Reply:
That table was moved from the https://github.com/zephyrproject-rtos/zephyr/wiki/Naming-Conventions. I don't know what was it's intent as I have not authred it. Based on your and @bjarki-trackunit comment I have decided to drop it.